lib/gateway/contentful_gateway.rb

```diff
@@ -1,3 +1,91 @@
 class Gateway::ContentfulGateway
-  def initialize(space_id:, access_token:); end
+  def initialize(space_id:, access_token:)
+    @space_id = space_id
+    @access_token = access_token
+    @client = Contentful::Client.new(
+      access_token: @access_token,
+      space: @space_id,
+      dynamic_entries: :auto,
+      raise_errors: true
+    )
+  end
+
+  def get_case_study(slug:)
+    case_study_response = @client.entries(
+      'content_type' => 'caseStudy',
+      'include' => 10,
+      'fields.slug' => slug
+    ).first
+
+    Domain::CaseStudy.new.tap do |case_study|
```

Comment:
Feels like this could be pulled out into a method `build_case_study_from_response` (or something similar)

Reply:
Makes sense! We've extracted a method in https://github.com/DFE-Digital/single-place-for-education-frontend/pull/6/commits/bd305ee69518781c00c92a5d6ae018b3206deeca. What do you think?